### DISCUSSION TITLE
About console and geoIP

### DISCUSSION THREAD

**Discussion by Captain-E** [akvorado/akvorado#879] [Q&A]
Hi. 
I have few questions.

This is my homepagewidgets config: 
**homepagetopwidgets:
  - src-port
  - dst-port
  - protocol
  - etype
  - src-country
  - dst-country**

Question 1: I tried to add "src-ip" or "dst-ip" widgets with no success. I read documentation again but "src-ip" widget is not mentioned there. Can we add src-ip?

Question 2: src-country and dst-country graphs are not displaying anything. Documentation says, by default, IPinfo databases are used. (https://ipinfo.io/). And last flow shows that it can distinguish between ip addresses. But graph is not displayed on dashboard. What should i do to display country graph?

![image](https://github.com/akvorado/akvorado/assets/76643470/44a181f8-0c1c-4828-a19d-408756f1b6ba)

Config: 
**geoip:
  optional: true
  asn-database: /usr/share/GeoIP/asn.mmdb
  geo-database: /usr/share/GeoIP/country.mmdb
  #asn-database: /usr/share/GeoIP/GeoLite2-ASN.mmdb
  #geo-database: /usr/share/GeoIP/GeoLite2-Country.mmdb**

Question 3: On the "visualize" tab, we can analyze flows and "average speed" of each flow. Is it possible to see "bytes" as well? (similar to elastiflow)
akvo screenshot:
![image](https://github.com/akvorado/akvorado/assets/76643470/c5d6a7a3-5bef-4972-8034-c4d7a9bba814)
elastiflow screenshot:
![image](https://github.com/akvorado/akvorado/assets/76643470/74936419-cf66-4e4d-bd0d-7b5f1b46bcec)

**Comment by vincentbernat** [comment]
Widget with directions need to have interface classification configured. For Dst, it adds `AND OutIfBoundary = 'external'` and for Src, it adds `AND InIfBoundary = 'external'`.
